packages/@glimmer-workspace/integration-tests/test/keywords/each-test.ts

```diff
@@ -30,6 +31,39 @@ class Each extends RenderTest {
 
     this.assertHTML('  ');
   }
+
+  @test
+  'each with array of tags is reactive per item'(assert: Assert) {
+    let array = [createUpdatableTag(), createUpdatableTag()];
+    let step = (index: number) => {
+      consumeTag(array[index]);
+      assert.step(String(index));
+      return index;
+    };
+
+    const Foo = defineComponent(
+      { step, array },
+      `{{#each array as |item index|}}
+        {{step index}}
+       {{/each}}`
+        .replaceAll(/^\s|\s+$|\s+(?=\s)/gu, '')
+        .replaceAll(/\n/gu, '')
+    );
+
+    this.renderComponent(Foo);
+    this.assertHTML(' 0  1 ');
+    assert.verifySteps(['0', '1']);
+
+    dirtyTag(array[0]!);
+    this.rerender();
+    this.assertHTML(' 0  1 ');
+    assert.verifySteps(['0']);
```

Comment:
I think steps count should be `0` (empty array), for following reasons: `item` not used inside block scope, so, there is no tag should be created for it, and because array is not changed, nothing should be recomputed. It may be recomputed in only one case: `item` used inside `step` and `step` consume tag. Like: `{{step item index}}`